python2/typing.py

```diff
@@ -563,6 +563,28 @@ def __subclasscheck__(self, cls):
 AnyStr = TypeVar('AnyStr', bytes, unicode)
 
 
+def _tp_cache(func, maxsize=128):
```

Comment:
A small comment: With this definition we could not later write:
```python
@_tp_cache(maxsize=200)
def __getitem__(self, parameters):
    ...
```
This causes an error (``function takes at least 1 argument...``). I would propose to either make maxsize a constant ``MAXSIZE`` or use the decorator with args pattern here.

Except for this small comment, everything LGTM.